### PR TITLE
hv_time: verify hv_time flag performance on windows guest

### PR DESCRIPTION
**Pull request by kimi1978wy**
Execute the gettime_cycles.exe to verify the flag "hv_time"

id: 1827459

Signed-off-by: Yu Wang <wyu@redhat.com>

### CODE REVIEW

**Comment by kimi1978wy** [comment]
(1/6) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.hv_time.q35: PASS (352.42 s)
 (2/6) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.hv_time.q35: PASS (371.54 s)
 (3/6) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.hv_time.q35: PASS (164.45 s)
 (4/6) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.hv_time.q35: PASS (189.63 s)
 (5/6) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.i386.1.io-github-autotest-qemu.hv_time.q35: PASS (488.14 s)
 (6/6) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.x86_64.1.io-github-autotest-qemu.hv_time.q35: PASS (502.15 s)

**Comment by kimi1978wy** [comment]
@YongxueHong Could you help to review it, thanks a lot.

**Comment by kimi1978wy** [comment]
@YongxueHong Could you help to review this patch? thanks a lot

File: qemu/tests/hv_time.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    flags_without_hv_time = params["flags_without_hv_time"]
+    flags_with_hv_time = params["flags_with_hv_time"]
+
+    error_context.context("Setting up environments")
+    _setup_environments()
+
+    error_context.context("Get cpu cycles without hv_time flag", logging.info)
+    cycles_without_flag = _get_cycles_with_flags(flags_without_hv_time)
+    vm = env.get_vm(params["main_vm"])
+    vm.graceful_shutdown(timeout=timeout)
```
Comment (YongxueHong) [inline qemu/tests/hv_time.py:137]: Suggest move line 136~137 into `_get_cycles_with_flags`.

File: qemu/tests/hv_time.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    check_pltclk_cmd = params["check_pltclk_cmd"]
+    gettime_filenames = params["gettime_filenames"].split()
+    copy_cmd = params["copy_cmd"]
+    delete_cmd = params["delete_cmd"]
+    run_gettime_cmd = params["run_gettime_cmd"]
+    timeout = params.get("timeout", 360)
+    flags_without_hv_time = params["flags_without_hv_time"]
+    flags_with_hv_time = params["flags_with_hv_time"]
+
+    error_context.context("Setting up environments")
```
[the remark this answers is not in the document]
Comment (kimi1978wy) [reply qemu/tests/hv_time.py:124]: updated, thanks

**Comment by kimi1978wy** [comment]
(1/2) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win8.i386.io-github-autotest-qemu.hv_time.q35: PASS (207.06 s)
 (2/2) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win8.x86_64.io-github-autotest-qemu.hv_time.q35: PASS (214.83 s)
 (1/7) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.hv_time.q35: PASS (215.36 s)
 (2/7) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.hv_time.q35: PASS (491.53 s)
 (3/7) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.hv_time.q35: PASS (414.57 s)
 (4/7) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.hv_time.q35: PASS (531.03 s)
 (5/7) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.hv_time.q35: PASS (250.72 s)
 (6/7) Host_RHEL.m8.u3.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.i386.1.io-github-autotest-qemu.hv_time.q35: PASS (234.85 s)
 (7/7) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.x86_64.1.io-github-autotest-qemu.hv_time.q35: PASS (210.25 s)

**Pull request merged**